src/rqt_publisher/publisher.py

```diff
@@ -365,6 +365,7 @@ def save_settings(self, plugin_settings, instance_settings):
             publisher_copy['enabled'] = False
             del publisher_copy['timer']
             del publisher_copy['message_instance']
+            self._node.destroy_publisher(publisher_copy['publisher'])
```

Comment:
Why should it be destroyed in `save_settings`? Why not in `clean_up_publishers`?